api/handlers.go

```diff
@@ -113,3 +114,54 @@ func (api *RecipeAPI) AddAllRecipeHandler(w http.ResponseWriter, req *http.Reque
 		}
 	}
 }
+
+//AddRecipeHandler - Add a Recipe
+func (api *RecipeAPI) AddRecipeHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	recipeID := vars["id"]
+	logD := log.Data{"recipe_id": recipeID}
+
+	// Read body
+	b, err := ioutil.ReadAll(req.Body)
+	defer req.Body.Close()
+	if err != nil {
+		log.Event(ctx, "error in reading request body", log.ERROR, log.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// Unmarshal to the shape of Response struct
+	var recipe recipe.Response
+	err = json.Unmarshal(b, &recipe)
+	if err != nil {
+		log.Event(ctx, "error returned from json unmarshal", log.ERROR, log.Error(err))
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
should this be a HTTP 400 Bad Request ?

Reply:
Ooooooo.....i don't know. In the previous PRs which has been merged, an `Internal server error` is given when there is a problem with the unmarshaling. In a way, it makes sense because it's something to do with unmarshaling which we can't really see. But that error occurs most of the time (100% for me) because of bad recipe given. So.....which one is the big question?

Reply:
I think maybe they are both appropriate in some scenarios. If existing implementations have used a 500 response I think it's fine 👍